pkg/api/v2/reporter/reporter.go

```diff
@@ -155,21 +155,28 @@ func (r *reporter) WriteReports(ctx context.Context, resourceErrs ResourceReport
 			merr = multierror.Append(merr, err)
 			continue
 		}
-		resources.UpdateMetadata(resource, func(meta *core.Metadata) {
-			meta.ResourceVersion = updatedResource.GetMetadata().ResourceVersion
-		})
-
-		logger.Debugf("wrote report %v : %v", updatedResource.GetMetadata().Ref(), status)
+		if updatedResource != nil {
+			resources.UpdateMetadata(resource, func(meta *core.Metadata) {
+				meta.ResourceVersion = updatedResource.GetMetadata().ResourceVersion
+			})
+			logger.Debugf("wrote report for %v : %v", updatedResource.GetMetadata().Ref(), status)
+		} else {
+			logger.Debugf("did not write report for %v : %v because resource was not found", resourceToWrite.GetMetadata().Ref(), status)
```

Comment:
in this case we should make sure that we remove `resource` from the `ResourceReports` map. to do this safely, we will need to copy the map before iterating over the copy, so we can remove from the original map